src/main/user-input-history-manager.ts

```diff
@@ -8,7 +8,9 @@ export class UserInputHistoryManager {
     }
 
     public addItem(userInput: string): void {
-        this.history.push(userInput);
+        if (this.history[this.history.length - 1] !== userInput) {
```

Comment:
What happens if `this.history` is an empty array? Doesn't it fail because it can't access the item with index `-1`?

Reply:
It would return `undefined` which is not equal to `userInput`, it works fine I tested it. I guess I can add one more check to see if the array is not empty.